gateway-js/src/index.ts

```diff
@@ -375,40 +317,89 @@ export class ApolloGateway implements GraphQLService {
       };
     }
 
-    // Before @apollo/gateway v0.23, ApolloGateway didn't expect stop() to be
-    // called after it started. The only thing that stop() did at that point was
-    // cancel the poll timer, and so to prevent that timer from keeping an
-    // otherwise-finished Node process alive, ApolloGateway unconditionally
-    // called unref() on that timeout. As part of making the ApolloGateway
-    // lifecycle more predictable and concrete (and to allow for a future where
-    // there are other reasons to make sure to explicitly stop your gateway),
-    // v0.23 tries to avoid calling unref().
-    //
-    // Apollo Server v2.20 and newer calls gateway.stop() from its stop()
-    // method, so as long as you're using v2.20, ApolloGateway won't keep
-    // running after you stop your server, and your Node process can shut down.
-    // To make this change a bit less backwards-incompatible, we detect if it
-    // looks like you're using an older version of Apollo Server; if so, we
-    // still call unref(). Specifically: Apollo Server has always passed an
-    // options object to load(), and before v2.18 it did not pass the `apollo`
-    // key on it. So if we detect that particular pattern, we assume we're with
-    // pre-v2.18 Apollo Server and we still call unref(). So this will be a
-    // behavior change only for:
-    // - non-Apollo-Server uses of ApolloGateway (where you can add your own
-    //   call to gateway.stop())
-    // - Apollo Server v2.18 and v2.19 (where you can either do the small
-    //   compatible upgrade or add your own call to gateway.stop())
-    // - if you don't call stop() on your ApolloServer (but in that case other
-    //   things like usage reporting will also stop shutdown, so you should fix
-    //   that)
-    const unrefTimer = !!options && !options.apollo;
```

Comment:
@glasser with `loadDynamic` gone does it seem correct that all this can go away? Does this need to be incorporated some way into my polling `SupergraphManager`s?

Reply:
I don't think the fact that we've refactored the code so that the timers are started in the managers means in and of itself that this can go away. Removing the unref is a behavior change for anyone using Apollo Server older than v2.18, in that ApolloServer.stop() won't call ApolloGateway.stop() and the event loop may be kept alive unnecessarily.

But v2.18 came out in September 2020 so I think we can happily remove this just as a no longer necessary workaround. The CHANGELOG should note that if you are using a version of Apollo Server older than v2.18, then server shutdown might not happen as smoothly in the next version of gateway; to fix this upgrade AS.